src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp

```diff
@@ -504,71 +504,165 @@ to_silo(const conduit::Node &field,
 }
 
 //-----------------------------------------------------------------------------
+// venn full -> sparse by element
 void
-multi_buffer_by_element_to_uni_buffer_by_element(const conduit::Node &matset,
-                                                 conduit::Node &dest,
+multi_buffer_by_element_to_uni_buffer_by_element(const conduit::Node &src_matset,
+                                                 const conduit::Node &src_field,
+                                                 conduit::Node &dest_matset,
+                                                 conduit::Node &dest_field,
                                                  const float64 epsilon)
 {
+    dest_matset.reset();
+    dest_field.reset();
+
     // set the topology
-    dest["topology"].set(matset["topology"]);
+    dest_matset["topology"].set(src_matset["topology"]);
+
+    if (src_field.dtype().is_object())
+    {
+        // copy over info from the old field
+        dest_field["association"].set(src_field["association"]);
+        dest_field["topology"].set(src_field["topology"]);
+        dest_field["matset"].set(src_field["matset"]);
+        dest_field["values"].set(src_field["values"]);
+    }
```

Comment:
Do we need to copy the values here? (Won't we over write later?)

Reply:
The only thing we write later is `matset_values`. We do not overwrite the `values`.